.github/workflows/cd.yml

```
@@ -0,0 +1,62 @@
+name: CD
+
+on: [push, pull_request]
```

It running `on.push` without specifying any branches might be overly broad. Is this deployment to PyPi supposed to be for only pushes to `main`? If so, it can be updated to:
```suggestion
on:
  push:
    branches:
      - main
    paths-ignore:
      - .github  # updates to CD infra shouldn't deploy to pypi
  pull_request:
    paths-ignore:
      - .github  # probably the case here, too
```
```